locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -2742,14 +2743,41 @@ protected LoginCredentials waitForWinRmAvailable(final ComputeService computeSer
             Callable<Boolean> checker = new Callable<Boolean>() {
                 public Boolean call() {
                     for (Map.Entry<WinRmMachineLocation, LoginCredentials> entry : machinesToTry.entrySet()) {
-                        WinRmMachineLocation machine = entry.getKey();
-                        WinRmToolResponse response = machine.executeCommand(
-                                ImmutableMap.of(WinRmTool.PROP_EXEC_TRIES.getName(), 1),
-                                ImmutableList.of("echo testing"));
-                        boolean success = (response.getStatusCode() == 0);
-                        if (success) {
-                            credsSuccessful.set(entry.getValue());
-                            return true;
+                        try {
+                            final WinRmMachineLocation machine = entry.getKey();
+                            WinRmToolResponse response = machine.executeCommand(
+                                    ImmutableMap.of(WinRmTool.PROP_EXEC_TRIES.getName(), 1),
+                                    ImmutableList.of("echo testing"));
+                            boolean success = (response.getStatusCode() == 0);
+                            if (success) {
+                                credsSuccessful.set(entry.getValue());
+                                Predicate<WinRmMachineLocation> pendingReboot = new Predicate<WinRmMachineLocation>() {
+                                    @Override
+                                    public boolean apply(@Nullable WinRmMachineLocation machine) {
+                                        WinRmToolResponse response = machine.executeCommand("echo test");
+                                        int statusCode = response.getStatusCode();
+                                        return statusCode == 0;
+                                    }
+                                };
+
+                                String verifyWindowsUp = getConfig(WinRmMachineLocation.VERIFY_WINDOWS_UP);
+                                if (Strings.isBlank(verifyWindowsUp) || verifyWindowsUp.equals("false")) {
+                                    return true;
+                                }
+
+                                Duration verifyWindowsUpTime = Duration.of(verifyWindowsUp);
+
+                                boolean restartHappened = Predicates2.retry(Predicates.not(pendingReboot), verifyWindowsUpTime.toSeconds(), 5, 60, TimeUnit.SECONDS).apply(machine);
```

Comment:
What type of exception is thrown when the predicate can't connect to the machine? `retry` will suppress only a specific set of exceptions, rethrowing the rest which will prevent the following code from executing. Suggest wrapping this in a try-catch and treating non-fatal exceptions as `restartHappened=true`.